pkg/reporting/util.go

```diff
@@ -87,3 +94,62 @@ func isDevEndpoint(endpoint string) bool {
 	result, _ := regexp.MatchString(`replicated-app`, endpoint)
 	return result
 }
+
+func AirgapReportSecret(name string, namespace string, kotsadmUID apimachinerytypes.UID, data []byte) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			// since this secret is created by the kotsadm deployment, we should set the owner reference
+			// so that it is deleted when the kotsadm deployment is deleted
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
Added this for consistency, but we don't really do this with other KOTS resources, so might not be necessary.

Reply:
It feels wrong from an architectural standpoint, as the IDs (instance and cluster IDs) are not tied to the `kotsadm` deployment like the case with the `replicated` sdk deployment. The cluster ID is stored in the `kotsadm-confg` configmap, and the instance ID is stored in rqlite, so the `kotsadm` deployment is ephemeral in that sense, and recreating it shouldn't cause this data to be deleted.

Another caveat is that the data will be lost during a migration from a `kotsadm` deployment to statefulset.

With that being said, we should figure out how to reset the data in case any of the IDs don't match, which can happen due to several reasons (re-installing the app, kots, etc...)

Reply:
Don't forget to enforce type checking for that new interface with something like:
```go
var _ Report = &InstanceReport{}
var _ Report = &PreflightReport{}
```

Reply:
> With that being said, we should figure out how to reset the data in case any of the IDs don't match, which can happen due to several reasons (re-installing the app, kots, etc...)

Might be a good use-case for another top-level key in the secret? Could store IDs there and ensure they match before appending data. Which IDs do we need to validate? License, Cluster, and Instance?

Reply:
Good idea, but we probably don't need top-level fields for that. How about just reading the IDs from the first event?

Reply:
Removed owner references, but leaving secrets as-is